### PR TITLE
New 311 route

### PR DESCRIPTION
**Pull request by cecilia-donnelly**
Add a new route, "/311."  Change the homepage to conditionally look different on this route.  Add a new logo to appear on that route.  Change names of the origins of requests, and the variables referring to them -- instead of `is_sms`, use origin, since there are now more than two origins.  Refer to the ones we now have as "sms," "web-home," and "chi-311-web."  More could come!

### CODE REVIEW

File: migrations/20170818-origin-rename.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/* Add copyright header here.
```
Comment (kfogel) [inline migrations/20170818-origin-rename.sql]: This is a trivial one-line file, so I'm not sure the copyright notice really matters, but just to model good behavior for others, let's put one in.

File: public/smokeAlarmPortal_files/base.css [hunk 2]
```diff
@@ -234,7 +234,7 @@ img.fourpanel {
 */
 
 #logo {
-  margin: 1em 0em;
+    margin: 1em 0em;
```
Comment (kfogel) [inline public/smokeAlarmPortal_files/base.css]: Unrelated whitespace change :-).

File: public/smokeAlarmPortal_files/base.css [hunk 3]
```diff
@@ -347,4 +347,10 @@ a.contrast-link {
     background-color: rgba(0, 0, 0, 0.3);
     -moz-border-radius: 5px 5px 5px 5px;
     border-radius: 5px 5px 5px 5px; /* standards-compliant: (IE) */
-}
\ No newline at end of file
+}
+
+.img-square {
```
Comment (kfogel) [inline public/smokeAlarmPortal_files/base.css]: This seems like an overly generic CSS class name for an image that we know is specifically meant to appear on a 311-route page to the right of the usual Red Cross logo.  Sure, we might add future 311 cities and they would have their own routes, but the CSS for this image is still 311-specific (as well as perhaps being square-specific).  Could the class have a name that indicates its function more closely?

File: views/admin/requests/index.js [hunk 4]
```diff
@@ -239,8 +239,8 @@ var getResults = function(callback) {
     };
 
     var createCSV = function() {
-        var requestFieldNames = ['id','name','address','city','state','zip','phone','email','date created','region'];
-        var requestFields = ['serial','name','address','city','state','zip','phone','email','createdAt','assigned_rc_region'];
+        var requestFieldNames = ['id','name','address','city','state','zip','phone','email','date created','region', 'origin'];
```
Comment (kfogel) [inline views/admin/requests/index.js]: The mismatch between "origin" in the field name and "source" in the request bugs me.  I guess that mismatch existed already, and shouldn't be fixed as part of this change, but would it be hard to fix afterwards in a separate change?

File: views/index.jade [hunk 5]
```diff
@@ -5,21 +5,24 @@ block content
     #top
       #logo.text-center
         img.logo(src='./smokeAlarmPortal_files/redcross-logo-white-letters.png', alt='American Red Cross')
+        if origin == '/311'
+          img.img-square(src='./images/311-chicago.png', alt='311 Chicago')
       #safety-message.row.text-center(role='complementary')
         p.col-xs-12.bigger-centered
           | #{__('Smoke alarms save lives.  Installing a smoke alarm is the first step to keeping your family safe.')}
-    #language
+      #language
```
Comment (kfogel) [inline views/index.jade]: Unrelated whitespace change.

File: views/index.jade [hunk 6]
```diff
@@ -5,21 +5,24 @@ block content
...
             a.language-toggle(data-lang='en') English
             | &nbsp;|&nbsp;
             a.language-toggle(data-lang='es') Español
     #main.row(role='main')
       #left.col-md-3
-        p
-          | #{__('Thank you for your interest in having FREE smoke alarms installed in your home.')} #{__('To request a FREE installation, please complete all fields in the form below and click “Submit.”')}
-        p
-          | #{__('After submitting your request, you will be contacted by an American Red Cross representative with information on installation availability in your area.')}
+        if origin != '/311'
```
Comment (kfogel) [inline views/index.jade]: Yup.  These conditionals all make sense.  Later, if/when we add more city 311 routes, then we'll need some kind of global conditional indicating "This is a 311 route for some city", but that won't be identical to the `origin`, excuse me I mean the `source` :-) that we're using.  Nothing needs to be changed now; just noting that a future generalization is likely.  If you think of a strategic place for a comment to that effect -- a place where someone making such a generalization change would encounter the comment early -- that might be good to put in.

File: views/index.js [hunk 7]
```diff
@@ -32,7 +32,14 @@ exports.saveRequest = function(req, res) {
     }).then( function(numRequests) {
         requestData = utils.getRequestData(req, numRequests, region_code);
         requestData = utils.createSerial(numRequests, requestData, region_code);
-        requestData.is_sms = 'web';
+        
+        // Check for 311 origin:
+        if (req.url == '/311') {
```
Comment (kfogel) [inline views/index.js]: The general-vs-specific thing here too -- again, I don't think we need to do anything about it right now, since we told Jim the route would be "/311" and we can future-proof it later (if more cities start doing this, then "/311" will just redirect to "/chi-311" or something, and no one will get the generic "/311" as their advertised URL anymore).

File: views/sms/index.js [hunk 8]
```diff
@@ -241,7 +241,7 @@ exports.respond = function(req, res) {
             return save_utils.countRequestsPerRegion(region_code);
         }).then( function(numRequests) {
             requestData = save_utils.createSerial(numRequests, req.cookies.request_object, region_code);
-            requestData.is_sms = 'sms';
+            requestData.origin = 'sms';
```
Comment (kfogel) [inline views/sms/index.js]: Oh, wait, I guess I was wrong?  This is the change where we are re-naming the request field from "is_sms" to "origin"?  In that case, let's use "origin" for both sides, or "source" for both sides.

File: views/utilities.js [hunk 9]
```diff
@@ -245,7 +245,7 @@ module.exports  = {
     saveRequestData: function(requestData) {
         return db.Request.create({
             name: requestData.name,
-            source: requestData.is_sms,
+            source: requestData.origin,
```
Comment (kfogel) [inline views/utilities.js]: Same here, obviously.

File: views/sms/index.js (hunk 8 above)
Comment (cecilia-donnelly) [inline views/sms/index.js]: Well, the "source" is the user-visible name and has always had that name.  I renamed the "is_sms" variable to "origin," as mentioned in the PR message, because that variable isn't just about whether or not the request came via SMS anymore.

I can make everything be `source`, sure.

File: migrations/20170818-origin-rename.sql (hunk 1 above)
Comment (cecilia-donnelly) [inline migrations/20170818-origin-rename.sql]: Oh right -- yeah, I took the header from another SQL file.  Who owns the copyright on this?

File: public/smokeAlarmPortal_files/base.css (hunk 3 above)
Comment (cecilia-donnelly) [inline public/smokeAlarmPortal_files/base.css]: +1

File: views/admin/requests/index.js (hunk 4 above)
Comment (cecilia-donnelly) [inline views/admin/requests/index.js]: (See [the other comment](https://github.com/redcross/smoke-alarm-portal/pull/232/files#r134554055), future readers!)

File: migrations/20170818-origin-rename.sql (hunk 1 above)
Comment (kfogel) [inline migrations/20170818-origin-rename.sql]: Yup -- same as on the code.

File: views/sms/index.js (hunk 8 above)
Comment (kfogel) [inline views/sms/index.js]: Whups, sorry, I missed that subtlety.  But yes, I think just making everything `source` would be best in the long run.

Comment (cecilia-donnelly) [inline views/sms/index.js]: Okay, actually, it's more complicated, and some of this can be done in this change and some of it should not.

There are a couple kinds of "origin" or "source," which is why this is so confusing -- thanks for catching it, @kfogel.  There's the conceptual source of the request -- did it come from an SMS, the regular web form, or the 311 web form?  Then there's the URL of the request, specifically, which pages use to determine which elements to display or not display.  Changing *that* is a bigger change, outside the scope of this PR. (I discovered that as I was trying to make it.)  But I think we can call the "conceptual" source the `source` everywhere, and I'll make that change in this PR. 👍

**Comment by cecilia-donnelly** [comment]
Thanks for the review, @kfogel -- I addressed these issues.  Merging now.
